tardis/tests/test_rpacket.py

```diff
@@ -0,0 +1,146 @@
+import os
+from sys import float_info as floats
+import random
+from ctypes import CDLL, c_double
+
+import pytest
+import numpy as np
+
+from tardis import __path__ as path
+
+test_path = os.path.join(path[0], 'montecarlo', 'montecarlo.so')
+MAX_FLOAT, MIN_FLOAT = floats[0], floats[3]
+MAX_INT = 10000
+MIN_INT = -10000
+
+tests = CDLL(test_path)
+SIZE = 10
+
+
+# Testing functions with float(C double) valued parameters
+np.random.seed(1)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu(double_value):
+	assert tests.test_rpacket_get_nu(double_value)
+
+np.random.seed(2)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_mu(double_value):
+	assert tests.test_rpacket_get_mu(double_value)
+
+np.random.seed(3)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_energy(double_value):
+	assert tests.test_rpacket_get_energy(double_value)
+
+np.random.seed(4)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_r(double_value):
+	assert tests.test_rpacket_get_r(double_value)
+
+np.random.seed(5)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_tau_event(double_value):
+	assert tests.test_rpacket_get_tau_event(double_value)
+
+np.random.seed(6)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_nu_line(double_value)
+
+np.random.seed(7)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_d_boundary(double_value)
+
+np.random.seed(8)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_d_electron(double_value)
+
+np.random.seed(9)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
```

Comment:
`test_rpacket_get_nu_line` was defined before two times. Have a look some of these are not quite correct.